src/network/networkd-link.c

```diff
@@ -1401,25 +1401,37 @@ static int link_set_bridge(Link *link) {
         if (r < 0)
                 return log_link_error_errno(link, r, "Could not append IFLA_PROTINFO attribute: %m");
 
-        r = sd_netlink_message_append_u8(req, IFLA_BRPORT_GUARD, !link->network->use_bpdu);
-        if (r < 0)
-                return log_link_error_errno(link, r, "Could not append IFLA_BRPORT_GUARD attribute: %m");
+        if (link->network->use_bpdu != -1) {
```

Comment:
i'd prefer if we'd write this as >= 0, i.e. accept any negative value as "unset", not just -1. Yes, we only ever set the value to -1, but I think it's generally nicer to distuingish only "negative", "positive" and "zero" for tri-state values...

Reply:
I had the same comment, but I resisted the urge to nit-pick ;)